src/wallet/wallet.cpp

```diff
@@ -2678,6 +2708,13 @@ void CWallet::IncrementNoteWitnesses(
         bool performOrchardWalletUpdates)
 {
     LOCK(cs_wallet);
+    int chainHeight = pindex->nHeight;
+
+    // Set the update cache flag.
+    int64_t prevWitCacheSize = nWitnessCacheSize;
+    if (nWitnessCacheSize < WITNESS_CACHE_SIZE) {
+        nWitnessCacheSize += 1;
+    }
```

Comment:
Maybe this is clearer?
```suggestion
    nWitnessCacheSize = std::min(nWitnessCacheSize + 1, WITNESS_CACHE_SIZE);
```